cg/src/domains/management/subjects/subjects.controller.js

```diff
@@ -7,7 +18,16 @@ class SubjectsController {
 
   async listSubjects(req, res) {
     const subjectsPage = await this.subjectsService.listSubjects(req.query.page);
-    res.json(subjectsPage);
+    if (subjectsPage.hasOwnProperty('error')) {
```

Comment:
I need to fix this line, although it's working, a subject can insert an 'error' key in his personal data and that would make his data not be blanked when displayed in the management dashboard. I can think of 2 fixes: 
- Reserve the error key to the system, not allowing a subject's personal data to have an  error key
- Do a more strict check, with the contents of the error field. this does not really solve the issue, but will make it less likely to the point that only someone trying to hack the system will be able to do it
I want to hear your opinions on this

Reply:
Why are we checking the output of the service layer for an error object at all? I would expect that any error case at the service layer is going to throw an error and then bubble out to the top level?

Reply:
This shows that I don't understand well how we handle errors in the project yet. I will try that out